### PR TITLE
DOCSP-34264 Add Oracle Prereq Content

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Task page for using Oracle with Relational Migrator / Data pre-reqs.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-34264/prerequisites/oracle/

## JIRA

https://jira.mongodb.org/browse/DOCSP-34264/

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65df6f9571111659822cb58f

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/prerequisites/oracle.txt [hunk 1]
```diff
@@ -10,4 +10,187 @@ Oracle Migration Prerequisites
...
+
+            #. Grant the required permissions to the service account:
+
+               .. code-block:: sql
+                  :copyable: true
+
+                  GRANT CREATE SESSION TO user;
+                  GRANT SELECT_CATALOG_ROLE TO user;
+                  GRANT SELECT ANY TABLE TO user;
+                  GRANT SELECT ON V$DATABASE TO user;
```
Comment (ianf-mongodb) [inline source/prerequisites/oracle.txt:60]: Is this specific to a version of Oracle?

```suggestion
                  GRANT SELECT ON V_$DATABASE TO user; 
```

**Review by davidsoulmdb — CHANGES_REQUESTED** [review]
Should mention that if you are on Amazon RDS, you will need to enable supplemental logging using https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.Oracle.CommonDBATasks.Log.html instead of by enabling LogMiner - note RDS is rare and not a priority to document

**Comment by davidsoulmdb** [comment]
GRANT SELECT ON V_$DATABASE TO user;

File: source/prerequisites/oracle.txt [hunk 2]
```diff
@@ -10,4 +10,187 @@ Oracle Migration Prerequisites
...
+                  :copyable: true
+
+                  CREATE USER 'user'@'localhost' IDENTIFIED BY 'password';
+
+            #. Grant the required permissions to the service account:
+
+               .. code-block:: sql
+                  :copyable: true
+
+                  GRANT SET CONTAINER TO user;
```
Comment (ianf-mongodb) [inline source/prerequisites/oracle.txt:119]: Anything in RDS Oracle will need to specified with:

```
CALL rdsadmin.rdsadmin_util.grant_sys_object('V_$DATABASE', '{{user}}', 'SELECT');
```

**Review by ianf-mongodb — COMMENTED** [review]
Updated.

**Comment by ianf-mongodb** [comment]
@davidsoulmdb I removed the RDS note per our discussion today

File: source/includes/fact-data-prep-oracle-step1.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+#. Grant select permissions to the service account:
+
+   .. code-block:: sql
+      :copyable: true
+
+      GRANT CREATE SESSION TO user;
+      GRANT SELECT_CATALOG_ROLE TO user;
+      GRANT SELECT ANY TABLE TO user;
+      GRANT SELECT ON V$DATABASE TO user;
```
Comment (poonam-meghnani) [inline source/includes/fact-data-prep-oracle-step1.rst:22]: This GRANT should be changed to 
`GRANT SELECT ON V_$DATABASE TO user;`

**Review by poonam-meghnani — COMMENTED** [review]
I tried the scripts using [Oracle EE 21c](https://container-registry.oracle.com/ords/f?p=113:4:4690747414344:::4:P4_REPOSITORY,AI_REPOSITORY,AI_REPOSITORY_NAME,P4_REPOSITORY_NAME,P4_EULA_ID,P4_BUSINESS_AREA_ID:9,9,Oracle%20Database%20Enterprise%20Edition,Oracle%20Database%20Enterprise%20Edition,1,0&cs=34X4kQvRYmjWswMOecnjv57vEGbg0Jnrj0qk4J6gj6nh6lAQKRYfMhAr5xtbdj5fe-IssmEIpJx08-sYJWRvsTA) containerised setup.  Most of the GRANTs looks good to me, however, with Oracle 12c and above, concept of pluggable databases was introduced. Though we assume that these scripts are taken care by a DBA who is aware of the below:
1) these scripts needs to be run using a SYSDBA role.
2) Some of these scripts can be run on PDB(plugable database)  while others like ARCHIVELOG related must be run on CDB(container/master database)
3) Supplemental logging is not allowed in Oracle Express editions.

I found this [Debezium blog](https://debezium.io/blog/2022/09/30/debezium-oracle-series-part-1/#configure-oracle-users) very useful in troubleshooting issues with redo logs as they are shared between PDB and CDB, not sure if it will be helpful to link it in docs.

File: source/includes/fact-data-prep-oracle-step1.rst (hunk 3 above)
Comment (poonam-meghnani) [reply source/includes/fact-data-prep-oracle-step1.rst:22]: @ianf-mongodb, I verified we don't really need the below for snapshot jobs, if the migration user is the owner of the migrating objects (tables), but only needed when the migration user is not owner and has least permissions. 
```
GRANT SELECT_CATALOG_ROLE TO user;
GRANT SELECT ANY TABLE TO user;
```
It does not need to be SELECT ANY TABLE (which is a wider privilege) but the below for each migrating table:

`GRANT SELECT ON TABLE TO user;`

Could we add -
1)  a note similar to - "you need these if you're not the table owner" for the above 2 statements?
2) Additionally, if the migration user is not owner, then below permission is also needed for snapshotting:
`GRANT FLASHBACK ON <TABLE> TO user;`

Comment (ianf-mongodb) [reply source/includes/fact-data-prep-oracle-step1.rst]: I have updated this step to include a query to check table owner ship and multiple code blocks for grants depending on if the service account running the sync job is owner or not.

**Review by ianf-mongodb — COMMENTED** [review]
Hi @poonam-meghnani thank you for the feedback and taking a look at this. I fixed the grant statement and added the additional content to the Learn More and About this Task section. Please LMK what you think about the edits. I also changed step 1 to include the select permissions depending on the ownership of the tables. TY!

**Comment by poonam-meghnani** [comment]
> Hi @poonam-meghnani thank you for the feedback and taking a look at this. I fixed the grant statement and added the additional content to the Learn More and About this Task section. Please LMK what you think about the edits. I also changed step 1 to include the select permissions depending on the ownership of the tables. TY!

Hello @ianf-mongodb, thanks for the quick changes, I think a few statements are redundant:
1) in Snapshot Jobs tab : Only first 2 statements are needed if service account is table owner, as the table owner shall have FLASHBACK grant on the owned tables already:
image

2) in Continuous Jobs tab: 

- Same changes as mentioned above in Snapshot jobs tab for FLASHBACK grant.
- GRANT FLASHBACK is redundant in Additional permissions: image
- Wherever we have ANY TABLE, we could change to <TABLE> to indicate selected tables participating in  migration.

WDYT?

**Review by davidsoulmdb — COMMENTED** [review]
Looks good to me

**Review by ianf-mongodb — COMMENTED** [review]
Thanks @poonam-meghnani your last requests have been updated in t[his commit ](https://github.com/mongodb/docs-relational-migrator/pull/92/commits/f3405c74663dc57232fe92f2c2a5fbd5beee0f99) I think this should be good to go!

File: source/includes/fact-data-prep-oracle-step1.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      GRANT SELECT ON V_$DATABASE TO <user>;
+
+   If the service account *is not* the table owner:
+
+   .. code-block:: sql
+      :copyable: true
+
+      GRANT CREATE SESSION TO <user>;
+      GRANT SELECT_CATALOG_ROLE TO <user>;
+      GRANT SELECT ANY TABLE TO <user>;
```
Comment (poonam-meghnani) [inline source/includes/fact-data-prep-oracle-step1.rst:43]: nit, `GRANT SELECT ON <table> TO <user>;`?

**Review by poonam-meghnani — APPROVED** [review]
Looks good to me now, just a small change to suggest more restrictive permissions, as some customers hesitate to give SELECT permissions on all tables.
Thank you for making suggested changes!

File: source/includes/fact-data-prep-oracle-step1.rst (hunk 4 above)
Comment (ianf-mongodb) [reply source/includes/fact-data-prep-oracle-step1.rst]: Done @poonam-meghnani !

**Pull request merged**